### ISSUE TITLE
tortoise: count votes for multiple block targets 

### ISSUE THREAD

**Issue by dshulyak** [spacemeshos/go-spacemesh#3608]
more info: https://community.spacemesh.io/t/deferred-block-fetching-a-k-a-dangling-pointers/309
depends on: https://github.com/spacemeshos/go-spacemesh/issues/3607

Each distinct combination of block id, layer id, tick height is a different voting target and weight is counted separately for every such target. The reason for doing so is because we don't know what is the "real" layer id and height is before we will fetch the block, but we want to fetch only blocks that are upvoted by majority.

- [ ] implementation plan. should include efficient way to represent and count votes for multiple targets

**Comment by dshulyak** [comment]
#### Multiple targets in tortoise
in-tortoise block data has to be extended from ballot votes (previously we didn't do that). there could be conflicting votes on the same id (e.g. different id, layer, height tuples). once we have full data we can verify which height and layer are true. lying about height and layer is a malfeasance behavior, but i will not rush with implementing that.

#### Tracking data availability
extend block info with boolean to track data availability. unless data is available - vote against the block, regardless of consensus. tortoise will not support block even if it was supported by hare without data available locally. `OnBlock` method updates data availability boolean to true, and must be called as it was before.

#### Reporting missing data availability for blocks in consensus
tortoise needs to make callers aware what block is in consensus but misses data.  Introduce InConsensus(from, to Layer) method, that copies blocks that are in consensus from internal tortoise state layer by layer preserving data availability boolean. this suppose to be called once per layer, so additional copy should not be a concern. caller will have to check data availability boolean before applying layer to vm. If block is not available - delegate to fetcher.

voting is not blocked by the aforementioned fetching. so if it is time to vote, and node doesn't have a block data - it votes against it.
